packages/eas-cli/src/devices/actions/create/inputMethod.ts

```diff
@@ -0,0 +1,166 @@
+import { AppleDeviceClass } from '../../../graphql/types/credentials/AppleDevice';
+import { AppleTeam } from '../../../graphql/types/credentials/AppleTeam';
+import log from '../../../log';
+import { confirmAsync, promptAsync } from '../../../prompts';
+import { isValidUDID } from '../../udids';
+
+interface DeviceData {
+  udid: string;
+  name?: string;
+  deviceClass: AppleDeviceClass | null;
+}
+
+const DEVICE_CLASS_DISPLAY_NAMES: Record<AppleDeviceClass, string> = {
+  [AppleDeviceClass.IPHONE]: 'iPhone',
+  [AppleDeviceClass.IPAD]: 'iPad',
+};
+
+export async function runInputMethodAsync(accountId: string, appleTeam: AppleTeam): Promise<void> {
+  log.newLine();
+  log(chalk.yellow('This is an advanced option. Use at your own risk.'));
+  log.newLine();
+
+  let registerNextDevice = true;
+  while (registerNextDevice) {
+    await collectDataAndRegisterDeviceAsync({ accountId, appleTeam });
+    log.newLine();
+    registerNextDevice = await confirmAsync({
+      message: 'Do you want to register another device?',
+    });
+  }
+}
+
+async function collectDataAndRegisterDeviceAsync({
+  accountId,
+  appleTeam,
+}: {
+  accountId: string;
+  appleTeam: AppleTeam;
+}): Promise<void> {
+  const { udid, deviceClass, name } = await collectDeviceDataAsync(appleTeam);
+
+  const spinner = ora(`Registering Apple Device on the Expo servers`).start();
+  try {
+    await AppleDeviceMutation.createAppleDeviceAsync(
+      {
+        appleTeamId: appleTeam.id,
+        identifier: udid,
+        name,
+        deviceClass: deviceClass ?? undefined,
+      },
+      accountId
+    );
+  } catch (err) {
+    spinner.fail();
+    throw err;
+  }
+  spinner.succeed();
+}
+
+async function collectDeviceDataAsync(
+  appleTeam: AppleTeam,
+  initialValues: Partial<DeviceData> = {}
+): Promise<DeviceData> {
+  const udid = await promptForUDIDAsync(initialValues.udid);
+  const name = await promptForNameAsync(initialValues.name);
+  const deviceClass = await promptForDeviceClassAsync(initialValues.deviceClass);
+  const deviceData: DeviceData = {
+    udid,
+    name,
+    deviceClass,
+  };
+
+  log.newLine();
+  log(
+    `We are going to register the following device in our database.
+This will ${chalk.bold('not')} register the device on the Apple Developer Portal yet.`
+  );
+  log.newLine();
+  printDeviceDataSummary(deviceData, appleTeam);
+  log.newLine();
+
+  const registrationConfirmed = await confirmAsync({
+    message: 'Is this what you want to register?',
+  });
+  if (!registrationConfirmed) {
+    log('No worries, just try again.');
+    log.newLine();
+    return await collectDeviceDataAsync(appleTeam, deviceData);
+  } else {
+    return deviceData;
+  }
+}
+
+async function promptForUDIDAsync(initial?: string): Promise<string> {
+  const { udid } = await promptAsync({
+    type: 'text',
+    name: 'udid',
+    message: 'UDID:',
+    initial,
+    validate: val => {
```

Comment:
Consider trimming whitespace (and possibly lowercasing/uppercasing the string based on length) to avoid copy paste / typing errors?